src/sentry/static/sentry/app/views/discover/sidebar/savedQueryList.tsx

```diff
@@ -26,55 +26,32 @@ type SavedQueriesProps = {
 type SavedQueriesState = {
   isLoading: boolean;
   data: SavedQuery[];
-  topSavedQuery: SavedQuery | null;
 };
 
 export default class SavedQueries extends React.Component<
   SavedQueriesProps,
   SavedQueriesState
 > {
-  constructor(props: SavedQueriesProps) {
-    super(props);
-    this.state = {
-      isLoading: true,
-      data: [],
-      topSavedQuery: props.savedQuery,
-    };
-  }
+  state: SavedQueriesState = {
+    isLoading: true,
+    data: [],
+  };
 
   componentDidMount() {
     this.fetchAll();
   }
 
-  componentWillReceiveProps(nextProps: SavedQueriesProps) {
-    // Refetch on deletion
-    if (!nextProps.savedQuery && this.props.savedQuery !== nextProps.savedQuery) {
+  componentDidUpdate(prevProps) {
+    // Re-fetch on deletion
+    if (prevProps.savedQuery && !this.props.savedQuery) {
       this.fetchAll();
     }
-
-    // Update query in the list with new data
-    if (nextProps.savedQuery && nextProps.savedQuery !== this.props.savedQuery) {
-      const data = this.state.data.map(savedQuery =>
-        savedQuery.id === nextProps.savedQuery!.id ? nextProps.savedQuery! : savedQuery
-      );
-      this.setState({data});
-    }
-
-    // Update saved query if any name / details have been updated
-    if (
-      nextProps.savedQuery &&
-      (!this.state.topSavedQuery ||
-        nextProps.savedQuery.id === this.state.topSavedQuery.id)
-    ) {
-      this.setState({
-        topSavedQuery: nextProps.savedQuery,
-      });
-    }
   }
 
   fetchAll() {
     fetchSavedQueries(this.props.organization)
       .then((data: SavedQuery[]) => {
+        data.sort((a, b) => (a.dateUpdated > b.dateUpdated ? -1 : 1));
```

Comment:
Why do we want to sort differently in the UI compared to the API response? I guess there's no pagination right now but this could end up being an issue in future.

Reply:
The saved query API now supports pagination but discover1 is bypassing pagination to maintain backwards compatibility.

Reply:
You should also be able to sort by date updated via endpoint parameters as well.